### PR TITLE
Edit view command

### PR DESCRIPTION
**Pull request by vishandi**
Current UI:
image

### CODE REVIEW

**Comment by vishandi** [comment]
Current UI:
image

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -155,6 +155,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### View Person Feature
+
+#### Implementation
+
+To make the GUI cleaner and simpler, we decided not to display the schedule of each person in the person list.
```
Comment (tyanhan) [inline docs/DeveloperGuide.md:162]: Would this part be better placed under design considerations rather than implementation?

File: src/main/java/seedu/address/ui/PersonViewCard.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import seedu.address.model.person.Person;
+import seedu.address.model.schedule.Schedule;
+
+/**
+ * An UI component that displays information of a {@code Person} in the viewed version.
+ */
+public class PersonViewCard extends UiPart<Region> {
```
Comment (tyanhan) [inline src/main/java/seedu/address/ui/PersonViewCard.java:18]: Would it be easier for people to understand if we call it `ScheduleCard` instead?

**Review by tyanhan — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (hanqinilnix) [reply docs/DeveloperGuide.md:162]: I agree with Yan Han on this point.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -155,6 +155,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+Instead, we made another panel on the right that will display the schedule of each person upon calling the function View.
+
+To do this, we made a new `FilteredList<Person>` object called `ViewedPersonList` in `ModelManager` to store the persons that we want to show.
+Moreover, we created `PersonViewPanel.java`, `PersonViewCard.java`, and their respective `.fxml` files so it will be 
+easier to maintain or develop.
+
+Overall, how this command works is similar to a combination of `delete` and `find`, in which we only take a particular index
+as input, and we filter the `ViewPersonList` by some criteria.
```
Comment (hanqinilnix) [inline docs/DeveloperGuide.md:170]:
```suggestion
Overall, how this command works is similar to a combination of `find` and `delete`, in which we only take an index as input, and we retrieve information based on the filtered list.
```
I think we should mention it based on the order we call the command. Also, filter by some criteria is not very clear.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -155,6 +155,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+3. `ViewCommand` creates a new `SamePersonPredicate` that returns `True` only if the tested `Person` equals to `personToView`.
+4. `ViewCommand` updates the `Model`'s `ViewedPersonList` by parsing in the `SamePersonPredicate`.
+5. `ViewCommand` constructs the `CommandResult` and returns it to the `LogicManager`.
+6. The GUI will be updated accordingly.
+
+<img src="images/ViewSequenceDiagram.png" />
+
+#### Design Considerations
+##### ViewedPersonList Object Type
+* **Alternative 1 (current choice):** FilteredList
```
Comment (hanqinilnix) [inline docs/DeveloperGuide.md:193]: I think we should describe the alternative a bit more. 
eg.
Alternative 1: Get information based on index of filtered list.
Alternative 2: ??
Actually.. I can't come up with an example, not sure what the consideration is.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -155,6 +155,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+    * Pros: Easier to implement, easier to develop if in the future we want to display more than one Person.
+    * Cons: Not intuitive since now the view command only support viewing one Person.
+* **Alternative 2:** Person
+    * Pros: More intuitive because it **is** the displayed Person.
+    * Cons: Can only view one Person at any time, need to change the implementation when developer wants to
+  display more than one Person.
+##### Displayed Attributes
+* **Alternative 1 (current choice):** Name, Phone, Schedule; No Schedule on Person List
```
Comment (hanqinilnix) [inline docs/DeveloperGuide.md:201]:
```suggestion
* **Alternative 1 (current choice):** Display required attributes like Name, Phone and Schedule
```

File: docs/diagrams/ViewSequenceDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+activate ViewCommand
+
+create SamePersonPredicate
+ViewCommand -> SamePersonPredicate : personToView
+activate SamePersonPredicate
+
+SamePersonPredicate --> ViewCommand : predicate
+deactivate SamePersonPredicate
+SamePersonPredicate -[hidden]-> ViewCommand
+destroy SamePersonPredicate
```
Comment (hanqinilnix) [inline docs/diagrams/ViewSequenceDiagram.puml:59]: Does the predicate end here if it is going to be used in the updateViewedPersonList()?

File: src/main/java/seedu/address/ui/PersonViewCard.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+    private static final String FXML = "PersonViewCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/ui/PersonViewCard.java:28]: We should take note of this. But I don't think it should be here, we should remove it.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 8]
```diff
@@ -124,17 +124,22 @@ public void setPerson(Person target, Person editedPerson) {
...
         filteredPersons.setPredicate(predicate);
     }
 
+    //=========== Filtered Person List Accessors =============================================================
+
+    /**
+     * Returns an unmodifiable view of the list of currently viewed person.
+     */
+    @Override
+    public ObservableList<Person> getViewedPersonList() {
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/model/ModelManager.java:139]: This is for the schedule card right? I think the method name should change to something better since it is not technically a person list. Maybe `getScheduleView()`?

File: src/main/java/seedu/address/ui/PersonViewPanel.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.person.Person;
+
+/**
+ * Panel containing the viewed list of persons.
+ */
+public class PersonViewPanel extends UiPart<Region> {
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/ui/PersonViewPanel.java:16]: Same as what Yan Han mentioned about. Perhaps ScheduleCardPanel?

**Review by hanqinilnix — CHANGES_REQUESTED** [review]

File: docs/diagrams/ViewSequenceDiagram.puml (hunk 6 above)
Comment (vishandi) [reply docs/diagrams/ViewSequenceDiagram.puml]: oh ya, missed that

File: src/main/java/seedu/address/ui/PersonViewPanel.java (hunk 9 above)
Comment (vishandi) [reply src/main/java/seedu/address/ui/PersonViewPanel.java:16]: Okay

File: src/main/java/seedu/address/model/ModelManager.java (hunk 8 above)
Comment (vishandi) [reply src/main/java/seedu/address/model/ModelManager.java:139]: I changed it to getViewSchedulePerson() cos I think the Person should be there.

**Review by hanqinilnix — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -155,6 +155,53 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### View Schedule Feature
+
+#### Implementation
+
+To implement `viewSchedule`, we made a new `FilteredList<Person>` object called `viewedPerson` in `ModelManager` to store the persons that we want to show.
+Moreover, we created `PersonViewPanel.java`, `PersonViewCard.java`, and their respective `.fxml` files so it will be
```
Comment (tyanhan) [inline docs/DeveloperGuide.md:163]: Remember to update the DG

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -155,6 +155,53 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### View Schedule Feature
+
+#### Implementation
+
+To implement `viewSchedule`, we made a new `FilteredList<Person>` object called `viewedPerson` in `ModelManager` to store the persons that we want to show.
```
Comment (tyanhan) [inline docs/DeveloperGuide.md:162]: I think we can omit the names of the objects themselves, since we want to explain the higher level details first before moving to the details later on. Otherwise the reader may be confused at the start as to what `viewSchedule` and `viewedPerson` is. Maybe we can just explain in layman terms, something like "To allow users to view their contacts' schedules, we made a new `FilteredList<Person>` object in `ModelManager` to store ...

**Review by tyanhan — CHANGES_REQUESTED** [review]
Slight changes in the DG, besides that LGTM!
